src/storage/src/statistics.rs

```diff
@@ -546,31 +647,298 @@ impl SinkStatistics {
+    /// ignored.
+    pub fn advance_global_epoch(&mut self, id: GlobalId) {
+        if self.worker_id == 0 {
+            self.global_source_statistics
+                .entry(id)
+                .and_modify(|(ref mut epoch, ref mut stats)| {
+                    *epoch += 1;
+                    stats.clear()
+                });
+            self.global_sink_statistics
+                .entry(id)
+                .and_modify(|(ref mut epoch, ref mut stats)| {
+                    *epoch += 1;
+                    stats.clear()
+                });
+        }
+    }
+
+    /// Re-initialize a source. If it already exists, then its local epoch is advanced.
+    pub fn initialize_source<F: FnOnce() -> SourceStatistics>(&mut self, id: GlobalId, stats: F) {
+        self.local_source_statistics
+            .entry(id)
+            .and_modify(|(ref mut epoch, ref mut stats)| {
+                *epoch += 1;
+                stats.clear()
+            })
+            .or_insert_with(|| (0, stats()));
+
+        if self.worker_id == 0 {
+            self.global_source_statistics
+                .entry(id)
+                .or_insert_with(|| (0, Default::default()));
+        }
+    }
+
+    /// Re-initialize a sink. If it already exists, then its local epoch is advanced.
+    pub fn initialize_sink<F: FnOnce() -> SinkStatistics>(&mut self, id: GlobalId, stats: F) {
+        self.local_sink_statistics
+            .entry(id)
+            .and_modify(|(ref mut epoch, ref mut stats)| {
+                *epoch += 1;
+                stats.clear()
+            })
+            .or_insert_with(|| (0, stats()));
+        if self.worker_id == 0 {
+            self.global_sink_statistics
+                .entry(id)
+                .or_insert_with(|| (0, Default::default()));
+        }
+    }
+
+    /// Ingest data from other workers.
+    pub fn ingest(
+        &mut self,
+        source_statistics: Vec<(usize, SourceStatisticsUpdate)>,
+        sink_statistics: Vec<(usize, SinkStatisticsUpdate)>,
+    ) {
+        // Currently, only worker 0 ingest data from other workers.
+        if self.worker_id == 0 {
+            for (epoch, stat) in source_statistics {
+                self.global_source_statistics.entry(stat.id).and_modify(
+                    |(global_epoch, ref mut stats)| {
+                        if epoch >= *global_epoch {
+                            stats.insert(stat.worker_id, stat);
+                        }
+                    },
+                );
+            }
+
+            for (epoch, stat) in sink_statistics {
+                self.global_sink_statistics.entry(stat.id).and_modify(
+                    |(global_epoch, ref mut stats)| {
+                        if epoch >= *global_epoch {
+                            stats.insert(stat.worker_id, stat);
+                        }
+                    },
+                );
+            }
+        }
+    }
+
+    fn _emit_local(
```

Comment:
What's going on with both `emit_local` and `_emit_local` existing?

Reply:
The wrapper doesn't emit anything on worker 0, to avoid broadcasting its statistics